### PR TITLE
Overtyped headlines for Index Alsos

### LINKED ISSUE #6173: Overtyped headlines for Index Alsos
Related issue: #6125 

**Is your feature request related to a problem? Please describe.**
In CPS indexes, journalists are able to overtype the headline and summary on a promo, to add context specific to that index. We currently do not support the headline overtype in Index Alsos.

**Describe the solution you'd like**
When we are rendering a promo that displays an Index Also, if that also has an overtyped headline, then we should use the overtyped value instead of the default value.

```
"headlines": {
                  "headline": "Impunzi zo mu Rwanda, u Burundi, Kongo... z’aba LGBT mu kaga i Nairobi",
                  "overtyped": "Impunzi zo mu Rwanda, u Burundi, Kongo... z’aba LGBT mu kaga i Nairobi "
                },
```

**Describe alternatives you've considered**
N/A

**Testing notes**
Update affected unit tests

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #6173

**Overall change:**
When rendering an Index Also, if the item has an `overtyped` field in the `headlines` object, then we should use the `overtyped` value instead of the `headline` value.

**Code changes:**
- Use the `overtyped` field if it is in the item object, otherwise use the `headline`
- Add assertion tests to check for the regular and overtyped headlines.

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added labels to this PR for the relevant pod(s) affected by these changes
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [X] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [X] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/StoryPromo/IndexAlsos/index.test.jsx [hunk 1]
```diff
@@ -23,21 +24,53 @@ describe('Index Alsos', () => {
...
+        <IndexAlsosContainer
+          alsoItems={relatedItems}
+          script={latin}
+          service="news"
+        />,
+      );
+
+      const firstListItem = container.querySelector('li');
+      const healine = firstListItem.getElementsByTagName('span')[2].innerHTML;
+      expect(healine).toEqual('APC ba ta isa ta kore ni ba – Buba Galadima');
```
Comment (Bopchy) [inline src/app/containers/StoryPromo/IndexAlsos/index.test.jsx]:
```suggestion
      const headline = firstListItem.getElementsByTagName('span')[2].innerHTML;
      expect(headline).toEqual('APC ba ta isa ta kore ni ba – Buba Galadima');
```

**Review by Bopchy — COMMENTED** [review]
Other than the minor suggested change, looks good 💯

**Review by thekp — APPROVED** [review]
Looks good to me 🏋️ Just need to apply Ruth's suggestions ^

**Review by tochwill — APPROVED** [review]

**Comment by PriyaKR** [comment]
Looks good to me.Adding `overtyped` key and value to the headlines in related items makes the overtyped headline to be used instead of headline.

**Pull request merged**